diff -Nur centrifuge.orig/indices/Makefile centrifuge/indices/Makefile
--- centrifuge.orig/indices/Makefile	2017-04-28 16:14:50.400228000 +0200
+++ centrifuge/indices/Makefile	2017-04-28 16:23:48.369641000 +0200
@@ -113,11 +113,11 @@
 p_compressed+h+v: export IDX_NAME:=p_compressed+h+v
 
 refseq_microbial: export COMPLETE_GENOMES:=archaea bacteria fungi protozoa
-refseq_microbial: export CHROMOSOME_LEVEL_GENOMES:=$(COMPLETE_GENOMES)
-refseq_microbial: export ANY_LEVEL_GENOMES:=viral
+##refseq_microbial: export CHROMOSOME_LEVEL_GENOMES:=$(COMPLETE_GENOMES)
+##refseq_microbial: export ANY_LEVEL_GENOMES:=viral
 ##refseq_microbial: export SMALL_GENOMES:=mitochondrion plasmid plastid # TODO
-refseq_microbial: export MAMMALIAN_TAXIDS:=9606 10090
-refseq_microbial: export INCLUDE_CONTAMINANTS:=1
+##refseq_microbial: export MAMMALIAN_TAXIDS:=9606 10090
+##refseq_microbial: export INCLUDE_CONTAMINANTS:=1
 refseq_microbial: export IDX_NAME:=refseq_microbial
 refseq_microbial: export CF_BUILD_OPTS+=--ftabchars 14
 
@@ -182,9 +182,9 @@
 $(REFERENCE_SEQUENCES_DIR):
 	mkdir -p $(REFERENCE_SEQUENCES_DIR)
 
-#$(TAXID_MAPS): | $(REFERENCE_SEQUENCES_DIR)
-#	rm $(patsubst %$(TAXID_SUFFIX),%.fna, $@)
-#	$(MAKE) -f $(THIS_FILE) $(patsubst %$(TAXID_SUFFIX),%.fna, $@)
+$(TAXID_MAPS): | $(REFERENCE_SEQUENCES_DIR)
+	rm $(patsubst %$(TAXID_SUFFIX),%.fna, $@)
+	$(MAKE) -f $(THIS_FILE) $(patsubst %$(TAXID_SUFFIX),%.fna, $@)
 
 nt.gz:
 	curl -o nt.gz ftp://ftp.ncbi.nih.gov/blast/db/FASTA/nt.gz
@@ -237,10 +237,10 @@
 	rm -rf $(TMP_DIR)
 endif
 
-$(REFERENCE_SEQUENCES_DIR)/all-%.fna: | $(REFERENCE_SEQUENCES_DIR) .dustmasker-ok
+$(REFERENCE_SEQUENCES_DIR)/all-%-chromosome_level.fna: | $(REFERENCE_SEQUENCES_DIR) .dustmasker-ok
 	[[ -d $(TMP_DIR) ]] && rm -rf $(TMP_DIR); mkdir -p $(TMP_DIR)
 	@echo Downloading and dust-masking $*
-	centrifuge-download -o $(TMP_DIR) $(CF_DOWNLOAD_OPTS) -d "$*" -P $(THREADS) refseq > \
+	centrifuge-download -o $(TMP_DIR) $(CF_DOWNLOAD_OPTS) -a "Chromosome" -d "$*" -P $(THREADS) refseq > \
 		$(TMP_DIR)/$(patsubst %.fna,%$(TAXID_SUFFIX),$(notdir $@))
 	cat $(TMP_DIR)/$*/*.fna > $@.tmp && mv $@.tmp $@
 	mv $(TMP_DIR)/$(patsubst %.fna,%$(TAXID_SUFFIX),$(notdir $@)) $(patsubst %.fna,%$(TAXID_SUFFIX),$@)
@@ -251,10 +251,10 @@
 	rm -rf $(TMP_DIR)
 endif
 
-$(REFERENCE_SEQUENCES_DIR)/all-%-chromosome_level.fna: | $(REFERENCE_SEQUENCES_DIR) .dustmasker-ok
+$(REFERENCE_SEQUENCES_DIR)/all-%.fna: | $(REFERENCE_SEQUENCES_DIR) .dustmasker-ok
 	[[ -d $(TMP_DIR) ]] && rm -rf $(TMP_DIR); mkdir -p $(TMP_DIR)
 	@echo Downloading and dust-masking $*
-	centrifuge-download -o $(TMP_DIR) $(CF_DOWNLOAD_OPTS) -a "Chromosome" -d "$*" -P $(THREADS) refseq > \
+	centrifuge-download -o $(TMP_DIR) $(CF_DOWNLOAD_OPTS) -d "$*" -P $(THREADS) refseq > \
 		$(TMP_DIR)/$(patsubst %.fna,%$(TAXID_SUFFIX),$(notdir $@))
 	cat $(TMP_DIR)/$*/*.fna > $@.tmp && mv $@.tmp $@
 	mv $(TMP_DIR)/$(patsubst %.fna,%$(TAXID_SUFFIX),$(notdir $@)) $(patsubst %.fna,%$(TAXID_SUFFIX),$@)
